hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -317,7 +321,8 @@ private boolean executeHttpOperation(final int retryCount,
 
       return false;
     } finally {
-      AbfsClientThrottlingIntercept.updateMetrics(operationType, httpOperation);
+      AbfsClientThrottlingIntercept instance = AbfsClientThrottlingInterceptFactory.getInstance(accountName, isAutoThrottlingEnabled, isSingletonEnabled);
+      instance.updateMetrics(operationType, httpOperation);
```

Comment:
if autothrottling is disabled , `AbfsClientThrottlingIntercept.getInstance`  will return null (see `AbfsClientThrottlingInterceptFactory.java`  L58).  That in turn will cause a NPE here and above at L286

Reply:
Added null checks